pycontrails/models/ps_model/ps_model.py

```diff
@@ -0,0 +1,936 @@
+        rocd_ms = units.ft_to_m(rocd) / 60
+        dv_dt = jet.acceleration(true_airspeed, dt_sec)
+        theta = jet.climb_descent_angle(true_airspeed, rocd_ms)
+
+        # Aircraft performance parameters
+        c_lift = lift_coefficient(atyp_param.wing_surface_area, aircraft_mass, air_pressure, mach_num, theta)
+        c_f = skin_friction_coefficient(rn)
+        c_drag_0 = zero_lift_drag_coefficient(c_f, atyp_param.psi_0)
+        e_ls = oswald_efficiency_factor(c_drag_0, atyp_param)
+        c_drag_w = wave_drag_coefficient(mach_num, c_lift, atyp_param)
+        c_drag = airframe_drag_coefficient(c_drag_0, c_drag_w, c_lift, e_ls, atyp_param.wing_aspect_ratio)
+
+        # Engine parameters and fuel consumption
+        f_thrust = thrust_force(aircraft_mass, c_lift, c_drag, dv_dt, theta)
+        c_t = engine_thrust_coefficient(f_thrust, mach_num, air_pressure, atyp_param.wing_surface_area)
+        eta = overall_propulsion_efficiency(mach_num, c_t, atyp_param)
+        fuel_flow = fuel_mass_flow_rate(
+            air_pressure, air_temperature, mach_num, c_t, eta,
+            atyp_param.wing_surface_area, q_fuel
+        )
+
+        if correct_fuel_flow:
+            fuel_flow = correct_fuel_flow(
+                fuel_flow, altitude_ft, air_temperature, air_pressure, mach_num,
+                atyp_param.ff_idle_sls, atyp_param.ff_max_sls
+            )
+        fuel_burn = jet.fuel_burn(fuel_flow, dt_sec)
+        return AircraftPerformanceData(
+            fuel_flow=fuel_flow,
+            aircraft_mass=aircraft_mass,
+            true_airspeed=true_airspeed,
+            thrust=f_thrust,
+            fuel_burn=fuel_burn,
+            engine_efficiency=eta,
+            rocd=rocd,
+        )
+
+
+# ----------------------
+# Atmospheric parameters
+# ----------------------
+
+
+def reynolds_number(
+        wing_surface_area: float,
+        mach_num: npt.NDArray[np.float_],
+        air_temperature: npt.NDArray[np.float_],
+        air_pressure: npt.NDArray[np.float_],
+) -> npt.NDArray[np.float_]:
+    """
+    Calculate Reynolds number
+
+    Parameters
+    ----------
+    wing_surface_area : float
+        Aircraft wing surface area, [:math:`m^2`]
+    mach_num : npt.NDArray[np.float_]
+        Mach number at each waypoint
+    air_temperature : npt.NDArray[np.float_]
+        Ambient temperature at each waypoint, [:math:`K`]
+    air_pressure: npt.NDArray[np.float_]
+        Ambient pressure, [:math:`Pa`]
+
+    Returns
+    -------
+    npt.NDArray[np.float_]
+        Reynolds number at each waypoint
+
+    References
+    ----------
+    - Refer to Eq. (3) of Poll & Schumann (2021).
+    - Poll & Schumann (2021). An estimation method for the fuel burn and other performance characteristics of civil
+        transport aircraft in the cruise. Part 1: fundamental quantities and governing relations for a general
+        atmosphere. Aero. J., 125(1284), 296-340, doi: 10.1017/aer.2020.62.
```

Comment:
Marc can help get these reference into the bibliography -- he just needs to know the DOI.